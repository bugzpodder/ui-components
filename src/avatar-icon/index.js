// @flow
export { AvatarIcon } from "./avatar-icon";
