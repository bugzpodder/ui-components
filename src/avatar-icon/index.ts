export { AvatarIcon } from "./avatar-icon";
