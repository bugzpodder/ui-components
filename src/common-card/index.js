// @flow
export { CommonCard } from "./card";
