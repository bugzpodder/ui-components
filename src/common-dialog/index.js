//@flow
export { CommonDialog } from "./dialog";
