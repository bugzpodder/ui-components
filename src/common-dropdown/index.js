// @flow
export { CommonDropdownMenu } from "./dropdown-menu";
