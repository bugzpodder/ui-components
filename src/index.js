// @flow
import type { PagedTableProps } from "./table";

export { Alert } from "./alert";
export { AvatarIcon } from "./avatar-icon";
export { CommonCard } from "./common-card";
export { BreastCancerRibbon } from "./logos";
export { CommonDialog, CommonMultiPageDialog } from "./common-dialog";
export { CommonDropdownMenu } from "./common-dropdown";
export { CommonPanel } from "./common-panel";
export { CommonMultiSelect, CommonSelect } from "./common-select";
export { CommonSuggest, SuggestionsOmniField } from "./common-suggest";
export { CommonSwitch } from "./common-switch";
export {
  CommonPage, CommonTabbedPage, CommonPageV2, CommonTabbedPageV2,
} from "./page";
export {
  DateValue,
  DateTimeValue,
  HumanizedDateTime,
  DateInput,
  DateTimeInput,
  wrapPickerUtilProvider,
  TimeInput,
} from "./date";
export { ExternalLink, ExternalLinkButton, LinkButton } from "./link";
export { Navbar } from "./__deprecated__/navbar";
export { NavbarV2 } from "./navbar";
export { NotificationCenter } from "./notification-center";
export { ReadOnlyTextField, ValidatedReadOnlyTextField } from "./readonly-text-field";
export { OmniChips, OmniSearchBar } from "./omni-search-bar";
export {
  SimpleTable, ExportTableButton, LargeSimpleTable, PagedTable,
} from "./table";
export { ExportButton, generateReport } from "./export-button";
export type { PagedTableProps };
export { SignInPage } from "./sign-in-page";
export { SpinnerOverlay } from "./spinner-overlay";
export { StyleWrapper } from "./style-wrapper";
export { TabbedTimelineCard, TimelineGraph, TimelineCard } from "./timeline-graph";
export { TextInput, TextInputV2 } from "./text-input";
export { TwoColumnGrid, TwoColumnRow } from "./two-column-grid";
export { UploadButton } from "./upload-button";
export { SelectionGrid } from "./selection-grid";
export {
  // $FlowFixMe: Property `readFileListAsTextArray` is missing in ModuleNamespace for export {}
  readFileListAsTextArray,
  // $FlowFixMe: Property `readFileListAsText` is missing in ModuleNamespace for export {}
  readFileListAsText,
  // $FlowFixMe: Property `readFileAsText` is missing in ModuleNamespace for export {}
  readFileAsText,
} from "./utils";
