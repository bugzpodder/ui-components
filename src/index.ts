export * from "./alert";
export * from "./avatar-icon";
export * from "./common-card";
export { BreastCancerRibbon } from "./logos";
export { CommonDialog, CommonMultiPageDialog } from "./common-dialog";
export { CommonDropdownMenu } from "./common-dropdown";
export * from "./common-panel";
export * from "./common-select";
export * from "./common-suggest";
export * from "./common-switch";
export * from "./page";
export * from "./date";
export * from "./link";
export { Navbar } from "./__deprecated__/navbar";
export * from "./navbar";
export * from "./notification-center";
export * from "./readonly-text-field";
export * from "./omni-search-bar";
export * from "./table";
export * from "./export-button";
export * from "./sign-in-page";
export * from "./spinner-overlay";
export * from "./style-wrapper";
export * from "./timeline-graph";
export * from "./text-input";
export * from "./two-column-grid";
export * from "./upload-button";
export * from "./selection-grid";
export * from "./utils";
export * from "./types/base-navbar";
export * from "./types/dropdown";
export * from "./types/paged-table";
export * from "./types/selection-grid";
export * from "./types/timeline";
export * from "./types/card";
export * from "./types/grid";
export * from "./types/panel";
export * from "./types/switch";
export * from "./types/dialog";
export * from "./types/notification";
export * from "./types/select";
export * from "./types/table";
export { FavoriteIconButton } from "./dev";
