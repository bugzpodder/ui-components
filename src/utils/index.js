// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
