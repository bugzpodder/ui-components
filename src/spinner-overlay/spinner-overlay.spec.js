// @flow
import React from "react";
import { render, Simulate } from "react-testing-library";
import Button from "@material-ui/core/Button";
import "jest-dom/extend-expect";
import { TestWrapper } from "@grail/components/src/utils";
import { SpinnerOverlay } from "./index";

test("render active spinner overlay", () => {
	const mockCallback = jest.fn();
	const { container, getByTestId } = render(
		<TestWrapper>
			<div>
				<Button onClick={mockCallback}>Active Spinner Test</Button>
				<SpinnerOverlay />
			</div>
		</TestWrapper>,
	);
	// TODO (jzhao/jsingh): jest does not recognize the overlay on the button and
	// simulating a click on the button here still works...
	// Simulate.click(getByText("Active Spinner Test"));
	// expect(mockCallback).not.toHaveBeenCalled();
	Simulate.click(getByTestId("spinner-overlay"));
	expect(container).toMatchSnapshot();
});

test("render inactive spinner overlay", () => {
	const mockCallback = jest.fn();
	const { container, getByText } = render(
		<TestWrapper>
			<div>
				<Button onClick={mockCallback}>Inactive Spinner Test</Button>
				<SpinnerOverlay isActive={false} />
			</div>
		</TestWrapper>,
	);
	Simulate.click(getByText("Inactive Spinner Test"));
	expect(mockCallback).toHaveBeenCalled();
	expect(container).toMatchSnapshot();
});
