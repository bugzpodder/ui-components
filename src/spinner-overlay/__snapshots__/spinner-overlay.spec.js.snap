// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render active spinner overlay 1`] = `
<div>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-flat"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Active Spinner Test
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="spinnerOverlay active"
      data-testid="spinner-overlay"
    >
      <div
        class="loading"
      />
    </div>
  </div>
</div>
`;

exports[`render inactive spinner overlay 1`] = `
<div>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-flat"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Inactive Spinner Test
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="spinnerOverlay"
      data-testid="spinner-overlay"
    >
      <div
        class="loading"
      />
    </div>
  </div>
</div>
`;
